Overhaul

Nats-ji opened the pull request:
Need test

Change list:
- File structure overhaul
- Added search bar
- Hotkeys now localizable
- Added favorites list
- Added hotkey list (bind any protocols with a hotkey)
- Brought back collapse #91 
- Auto update version number in manifest.json

![image](https://user-images.githubusercontent.com/13299626/108668067-625da780-7515-11eb-918c-3ff84567cc67.png)

Commit:
Move updateLang() to a separate file

Commit:
Remove redundant lines

Commit:
Modularize hotkeys

Commit:
Remove redundant codes

Commit:
Modularize options

Commit:
Modularize ui

Commit:
Move onOverlayOpen/Close to ui

Commit:
Clean up require

Commit:
Move get fact value function to fact.lua

Commit:
Update update.lua

Commit:
Add i18n support for hotkeys

Commit:
Add search bar

Commit:
Update language scan script

Commit:
Modularize ui list

Commit:
update.lua now replace old language file

Commit:
Update language files

Commit:
Add [BD] prefix to console output

Commit:
Merge pull request #92 from Nats-ji/overhaul

Overhaul

Commit:
Update da.lua

Nats-ji is a hero

Commit:
Merge pull request #93 from VP-EN/overhaul-da

Update da.lua

Commit:
Add custom widgets

Commit:
Add fav and hk button

Commit:
Add fav list

Commit:
Add hk list

Commit:
Add favorites.json

Commit:
Update language

Commit:
Merge branch 'overhaul' into overhaul

Commit:
Merge pull request #95 from Nats-ji/overhaul

Overhaul

Commit:
Update README.md

Commit:
Update README.md

Commit:
Auto update version in manifest.json

Commit:
Update manifest.json

Commit:
Update README.md

Update readme with a quick custom hotkey "how to" support

Commit:
Update README.md

Commit:
Merge pull request #97 from VP-EN/overhaul-readme

Update readme with a quick custom hotkey "how to" support

VP-EN reviewed (approved):
Looks absolutely amazing :) Great work Nats-ji 👍 
I've added a short / quick how-to configure the hotkeys, incase someone doesn't get it.

I've tested pretty much all I can ingame, and everything seems to work.

Commit:
Update README.md

Commit:
Update README.md

small changes, formatting

Commit:
Update README.md

Added a few GIFs to show:
- Preview of features (all sub-menus)
- The favorites function
- The search function
- How to add and remove custom hotkeys

Commit:
Update README.md

image host fix

Commit:
Update README.md

Commit:
Update README.md

Commit:
Merge pull request #98 from VP-EN/overhaul-readme

Readme update with GIFs

ghost commented:
Updated the README with a little preview section and updated the how-to guide on custom hotkeys.
I say this is ready to merge 👍

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

less compressed gif, think I'm done making these small changes now -.-

osulli commented on mods/braindance_protocol/utility.lua:
```diff
@@ -1,11 +1,11 @@
 
 function Utilities.StartProtocol(name)
-    print(string.format("Executing Protocol: %s", name))
+    print(string.format("[BD] Executing Protocol: %s", name))
```
I'm wondering if we can deprecate this as mentioned in #64 ?

osulli reviewed (approved):
Looks great

Commit:
Update README.md

Commit:
Remove StartProtocol()

Commit:
Merge remote-tracking branch 'upstream/overhaul' into overhaul

Commit:
Merge pull request #99 from Nats-ji/overhaul

Removed StartProtocol()
